AppLovin MAX Demo App - Java/app/src/main/java/com/applovin/enterprise/apps/demoapp/ads/max/mrecs/RecyclerViewMrecAdActivity.java

```diff
@@ -0,0 +1,250 @@
+
+                @Override
+                public void onAdCollapsed(final MaxAd maxAd) { }
+
+                @Override
+                public void onAdDisplayed(final MaxAd maxAd) { /* DO NOT USE - THIS IS RESERVED FOR FULLSCREEN ADS ONLY AND WILL BE REMOVED IN A FUTURE SDK RELEASE */ }
+
+                @Override
+                public void onAdHidden(final MaxAd maxAd) { /* DO NOT USE - THIS IS RESERVED FOR FULLSCREEN ADS ONLY AND WILL BE REMOVED IN A FUTURE SDK RELEASE */ }
+            } );
+
+            // Set this extra parameter to work around SDK bug that ignores calls to stopAutoRefresh()
+            adView.setExtraParameter( "allow_pause_auto_refresh_immediately", "true" );
+            adView.stopAutoRefresh();
+
+            // Load the ad
+            adView.loadAd();
+            adViews.add( adView );
+        }
+    }
+
+    private enum HolderViewType
+    {
+        AD_VIEW,
+        CUSTOM_VIEW
+    }
+
+    public class CustomRecyclerAdapter
+            extends RecyclerView.Adapter
+    {
+        private final LayoutInflater inflater;
+        private final List<String>   data;
+
+        public CustomRecyclerAdapter(Activity activity, List<String> data)
+        {
+            this.inflater = activity.getLayoutInflater();
+            this.data = data;
+        }
+
+        @NonNull
+        @Override
+        public RecyclerView.ViewHolder onCreateViewHolder(@NonNull final ViewGroup parent, final int viewType)
+        {
+            HolderViewType holderViewType = HolderViewType.values()[viewType];
+
+            switch ( holderViewType )
+            {
+                case AD_VIEW:
+                    return new AdViewHolder( inflater.inflate( R.layout.activity_mrec_ad_view_holder, parent, false ) );
+                case CUSTOM_VIEW:
+                    return new CustomViewHolder( inflater.inflate( R.layout.activity_mrec_custom_view_holder, parent, false ) );
+                default:
+                    return null;
+            }
+        }
+
+        @Override
+        public void onBindViewHolder(@NonNull final RecyclerView.ViewHolder holder, final int position)
+        {
+            if ( holder instanceof AdViewHolder )
+            {
+                // Select an ad view to display
+                MaxAdView adView = adViews.get( ( position / AD_INTERVAL ) % AD_VIEW_COUNT );
+
+                // Configure cell with an ad
+                ( (AdViewHolder) holder ).configureWithAdView( adView );
+            }
+            else if ( holder instanceof CustomViewHolder )
+            {
+                ( (CustomViewHolder) holder ).textView.setText( data.get( position ) );
+            }
+        }
+
+        @Override
+        public void onViewRecycled(@NonNull final RecyclerView.ViewHolder holder)
+        {
+            super.onViewRecycled( holder );
+
+            ViewGroup viewGroup = (ViewGroup) holder.itemView;
+
+            for ( int i = viewGroup.getChildCount(); i >= 0; i-- )
```

Comment:
Why are we iterating over all the views. We already have the `adView` reference from `AdViewHolder` below right. We just need to call `viewGroup.removeView( View )`

Reply:
@alacau we should also update the code on iOS to do the same